ifcb-parsing-processing/README.md

```diff
@@ -0,0 +1,117 @@
+
+**Installation**
+
+1) Clone the repo:
+
+`git clone [repo URL]`
+
+2) Set up a Python environment:
+
+`python -m venv venv
+source venv/bin/activate  # On macOS/Linux
+venv\Scripts\activate     # On Windows`
+
+3) Install dependencies:
+
+`pip install -r ifcb-parsing-processing/requirements.txt`
+
+4) Launch the Jupyter Notebooks (nativate to the IFCB_Log_Notebooks directory and launch Jupyter)
+
+`cd IFCB_Log_Notebooks
+jupyter notebook`
+
+
+
+#### Other Python file background/context
+
+**make_csv_from_hdr_parse & underway_data_processing**
+
+These scripts were developed during the first deployment and shipboard ops of the OOI IFCBs. They have since been superseded by the scripts in the Jupyter Notebooks within IFCB_Log_Notebooks.
+
+1. hdr_roi_compare.py 
+
+      used to verify all corrisponding ROI and HDR files are in a directory. Outputs a csv files containing columns for general file name, ROI file name (including extension), and HDR file name (including extension)
+      this is independant of the processing order technically; i.e., this doesn't create or cause any dependancies but it can be used as a pre-process qc step
+
+   
+3. gather_variables_from_hdr.py
+
+      gathers file_parameters = ["FileComment", "triggerCount", "roiCount", "runTime", "inhibitTime", "humidity", "temperature", "runTime", "PMTAhighVoltage", "PMTBhighVoltage", "humidity", "SyringeSampleVolume", "syringeSamplingSpeed", "temperature", "RunFastFactor", "sampleNumber", "runSampleFast"]
+      from hdr files
+
+      outputs a csv where each gathered param is a column and each row represents one hdr file from the designated directory/ies of ifcb data (e.g., hdrFileSummary_2024-08-14.csv)
+
+      this output is used in the hdrSummaryProcessing.py file
+
+4. hdrSummaryProcessing.py
+
+      outputs a csv containing
+   
+       - Datetime pulled from the file name
+   
+       - lookTime (lookTime = runTime - inhibitTime)
+   
+       - runSampleFast_Int
+   
+       - water_sample_type (in-lab testing, underway, discrete_sample, etc.; taken from file comment)
+   
+       - flowRate_mins (df['SyringeSampleVolume'] / df['syringeSamplingSpeed'])
+   
+       - volumeAnalyzed (df['RunFastFactor'] * df['runSampleFast_Int']) * df['flowRate_mins'] * (df['lookTime']/60)/5
+
+       output csv filename example = hdrFileSummaryFULL_2024-08-14.csv
+
+6. summaryHDR_and_underway.py
+
+    merges hdrFileSummaryFULL csv with merged underway ship data.
+
+
+
+#### OOI IFCB Dashboard background/context 
+
+- 1 dashboard is used to provide all OOI IFCB data
+- Each dataset in the public-facing dashboard represents 1 array (currently there will be 1 total dataset representing Pioneer MAB, this will contain moored, discrete, and underway data from all OOI IFCBs that operate at the Pioneer MAB site)
```

Comment:
This needs to be updated for the separate MAB moored and MAB shipboard datasets.